source/frontend/src/features/mapSideBar/acquisition/tabs/agreement/common/UpdateAcquisitionAgreementView.tsx

```diff
@@ -0,0 +1,118 @@
+import axios, { AxiosError } from 'axios';
+import { Formik } from 'formik';
+import styled from 'styled-components';
+
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import SidebarFooter from '@/features/mapSideBar/shared/SidebarFooter';
+import { getCancelModalProps, useModalContext } from '@/hooks/useModalContext';
+import { IApiError } from '@/interfaces/IApiError';
+import { ApiGen_Concepts_Agreement } from '@/models/api/generated/ApiGen_Concepts_Agreement';
+
+import AcquisitionAgreementForm from '../form/AcquisitionAgreementForm';
+import { AcquisitionAgreementFormYupSchema } from '../form/AcquisitionAgreementFormYupSchema';
+import { AcquisitionAgreementFormModel } from '../models/AcquisitionAgreementFormModel';
+
+export interface IUpdateAcquisitionAgreementViewProps {
+  isLoading: boolean;
+  initialValues: AcquisitionAgreementFormModel | null;
+  onSave: (agreement: ApiGen_Concepts_Agreement) => Promise<ApiGen_Concepts_Agreement | undefined>;
+  onSuccess: () => void;
+  onError: (e: AxiosError<IApiError>) => void;
+  onCancel: () => void;
+}
+
+const UpdateAcquisitionAgreementView: React.FunctionComponent<
+  React.PropsWithChildren<IUpdateAcquisitionAgreementViewProps>
+> = ({ isLoading, initialValues, onSave, onSuccess, onError, onCancel }) => {
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const cancelFunc = (resetForm: () => void, dirty: boolean) => {
+    if (!dirty) {
+      resetForm();
+      onCancel();
+    } else {
+      setModalContent({
+        ...getCancelModalProps(),
+        handleOk: () => {
+          resetForm();
+          setDisplayModal(false);
+          onCancel();
+        },
+      });
+      setDisplayModal(true);
+    }
+  };
+
+  return (
+    initialValues && (
+      <StyledFormWrapper>
+        <Formik<AcquisitionAgreementFormModel>
+          enableReinitialize
+          initialValues={initialValues}
+          validationSchema={AcquisitionAgreementFormYupSchema}
+          onSubmit={async (values, formikHelpers) => {
+            try {
+              const agreementSaved = await onSave(values.toApi());
+              if (agreementSaved) {
+                onSuccess();
+              }
+            } catch (e) {
+              if (axios.isAxiosError(e)) {
+                const axiosError = e as AxiosError<IApiError>;
+                onError && onError(axiosError);
+              }
+            } finally {
```

Comment:
the form should not 'know' that the save is going through an axios call. Also, this should be done by the container instead.

Reply:
updated